pkg/image/admission/admission.go

```diff
@@ -0,0 +1,319 @@
+		if om.Namespace == "" {
+			glog.V(5).Infof("%s: namespace was not set for object, defaulting to %s", pluginName, attr.GetNamespace())
+			om.Namespace = attr.GetNamespace()
+		}
+	}
+
+	return a.limitRanger.Admit(attr)
+}
+
+// SupportsAttributes is a helper that returns true if the resource is supported by the plugin.
+// Implements the LimitRangerActions interface.
+func (a *imageLimitRangerPlugin) SupportsAttributes(attr kadmission.Attributes) bool {
+	if attr.GetSubresource() != "" {
+		return false
+	}
+
+	resource := attr.GetResource()
+	return resource == imageapi.Resource("imagestreammappings") ||
+		resource == imageapi.Resource("imagestreamtags") ||
+		resource == imageapi.Resource("imagestreams")
+}
+
+// SupportsLimit provides a check to see if the limitRange is applicable to image objects.
+// Implements the LimitRangerActions interface.
+func (a *imageLimitRangerPlugin) SupportsLimit(limitRange *kapi.LimitRange) bool {
+	if limitRange != nil {
+		for _, limit := range limitRange.Spec.Limits {
+			if limit.Type == imageapi.LimitTypeImageSize {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// Limit is the limit range implementation that checks resource against the
+// image limit ranges.
+// Implements the LimitRangerActions interface
+func (a *imageLimitRangerPlugin) Limit(limitRange *kapi.LimitRange, kind string, obj runtime.Object) error {
+	var images map[string]imageapi.Image
+
+	switch obj.(type) {
+	case *imageapi.ImageStreamMapping:
+		ism, ok := obj.(*imageapi.ImageStreamMapping)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStreamMapping: %#v", pluginName, obj)
+			return nil
+		}
+		images = map[string]imageapi.Image{ism.Image.Name: ism.Image}
+	case *imageapi.ImageStreamTag:
+		tag, ok := obj.(*imageapi.ImageStreamTag)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStreamTag: %#v", pluginName, obj)
+			return nil
+		}
+
+		if tag.Tag == nil || tag.Tag.From == nil {
+			glog.V(4).Infof("%s: ignoring ist with no tag reference to follow: %v", pluginName, tag)
+			return nil
+		}
+
+		computer := imagequota.NewGenericImageStreamUsageComputer(a.osClient, true, false)
+		ref, err := computer.GetImageReferenceForObjectReference(tag.Namespace, tag.Tag.From)
+		if err != nil {
+			return err
+		}
+		image, err := computer.GetImage(ref.ID)
+		if err != nil {
+			return err
+		}
+		if image == nil {
+			return fmt.Errorf("unable to find image with ID %s", ref.ID)
+		}
+		images = map[string]imageapi.Image{image.Name: *image}
+	case *imageapi.ImageStream:
+		is, ok := obj.(*imageapi.ImageStream)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStream: %#v", pluginName, obj)
+			return nil
+		}
+
+		previousIS, _, err := a.imageStreamStore.Get(is)
+		if err != nil {
+			// don't return, process all images in the IS
+			glog.V(4).Infof("%s: error retrieving previous image stream from cache: %v", pluginName, err)
+		}
+
+		if previousIS == nil {
```

Comment:
Wouldn't be safer to check here the `existence` return value rather than `previousIS`?